src/librustc_codegen_llvm/intrinsic.rs

```diff
@@ -141,6 +144,38 @@ pub fn codegen_intrinsic_call(
             let llfn = cx.get_intrinsic(&("llvm.debugtrap"));
             bx.call(llfn, &[], None)
         }
+        "va_arg" => {
+            match fn_ty.ret.layout.abi {
+                layout::Abi::Scalar(ref scalar) => {
+                    match scalar.value {
+                        Primitive::Int(..) => {
+                            if cx.size_of(ret_ty).bytes() < 4 {
+                                // va_arg should not be called on a integer type
```

Comment:
Added the logic to manually promote a smaller type to an `i32` and then truncate it back. This should fix the windows issue seen in the CI.

Note: I still don't have my full dev setup, so I cannot test this at the moment.